firedrake/interpolation.py

```diff
@@ -254,20 +307,40 @@ def _interpolator(V, tensor, expr, subset, arguments, access):
                 raise NotImplementedError("Can only interpolate Python kernels with Lagrange elements")
             pts, = dual.pt_dict.keys()
             to_pts.append(pts)
-        kernel, oriented, needs_cell_sizes, coefficients = compile_python_kernel(expr, to_pts, to_element, V, coords)
+        kernel, oriented, needs_cell_sizes, coefficients = compile_python_kernel(expr, to_pts, to_element, V, source_mesh.coordinates)
         first_coeff_fake_coords = False
     else:
         raise RuntimeError("Attempting to evaluate an Expression which has no value.")
 
-    cell_set = coords.cell_set
+    cell_set = target_mesh.cell_set
     if subset is not None:
         assert subset.superset == cell_set
         cell_set = subset
     parloop_args = [kernel, cell_set]
 
     if first_coeff_fake_coords:
-        # Replace with real coords coefficient
-        coefficients[0] = coords
+        # Replace with real source mesh coordinates
+        coefficients[0] = source_mesh.coordinates
+
+    if trans_mesh:
```

Comment:
(Suggested) `if target_mesh is not source_mesh`:

Reply:
agreeed